### PR TITLE
feat: Change story map style

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Added map style selector to story maps

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
- https://github.com/techmatters/terraso-web-client/issues/1658

### Verification steps

Screenshot 2024-04-12 at 10 29 42 AM
Screenshot 2024-04-12 at 10 29 38 AM

### CODE REVIEW

**Review by paulschreiber — CHANGES_REQUESTED** [review]
- Remove the confirmation dialog — change immediately.
- Switching layers give 404 errors — please confirm the style URLs and that the styles are public.
- Satellite and detailed satellite are too similar. Once I can see what they actually are, I will give better name suggestions.

File: src/storyMap/components/StoryMapForm/MapLocationDialog.js [hunk 1]
```diff
@@ -179,13 +201,22 @@ const MapLocationDialog = props => {
   }, [location, config.chapters, config.titleTransition?.location, chapterId]);
 
   const handleConfirm = useCallback(() => {
-    onConfirm({
+    const location = _.omitBy(_.isNil, {
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/storyMap/components/StoryMapForm/MapLocationDialog.js:204]: @paulschreiber Just null or undifined, some of these values are numbers so the could include the `0` falsy value

**Comment by josebui** [comment]
> Switching layers give 404 errors — please confirm the style URLs and that the styles are public.

@paulschreiber Where do you see that error?, seems to be working locally and on staging for me

**Comment by josebui** [comment]
> Remove the confirmation dialog — change immediately.

@paulschreiber Can you add more context about this?

File: src/storyMap/components/StoryMapForm/MapLocationDialog.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulschreiber) [reply src/storyMap/components/StoryMapForm/MapLocationDialog.js:184]: What about `false`?

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**